sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -2284,6 +2369,111 @@ private static <T> Mono<T> getPointOperationResponseMonoWithE2ETimeout(
         return rxDocumentServiceResponseMono;
     }
 
+    private <T> Mono<T> handleCircuitBreakingFeedbackForPointOperation(
+        Mono<T> response,
+        AtomicReference<RxDocumentServiceRequest> requestReference) {
+
+        return response
+            .doOnSuccess(ignore -> {
+
+                if (this.globalPartitionEndpointManagerForCircuitBreaker.isPartitionLevelCircuitBreakingApplicable(requestReference.get())) {
+                    RxDocumentServiceRequest succeededRequest = requestReference.get();
+                    checkNotNull(succeededRequest.requestContext, "Argument 'succeededRequest.requestContext' must not be null!");
+
+                    PointOperationContextForCircuitBreaker pointOperationContextForCircuitBreaker = succeededRequest.requestContext.getPointOperationContextForCircuitBreaker();
+                    checkNotNull(pointOperationContextForCircuitBreaker, "Argument 'pointOperationContextForCircuitBreaker' must not be null!");
+                    pointOperationContextForCircuitBreaker.setHasOperationSeenSuccess();
+
+                    this.globalPartitionEndpointManagerForCircuitBreaker.handleLocationSuccessForPartitionKeyRange(succeededRequest);
+                }
+            })
+            .doOnError(throwable -> {
+                if (throwable instanceof OperationCancelledException) {
+
+                    if (this.globalPartitionEndpointManagerForCircuitBreaker.isPartitionLevelCircuitBreakingApplicable(requestReference.get())) {
+                        RxDocumentServiceRequest failedRequest = requestReference.get();
```

Comment:
Nice catch - can we sync up on this on Monday? I don't think bubbling up `NullPointerException` is right then - since it is a side-effect and not the real issue.